javascripts/discourse/components/discourse-tag-banners.js

```diff
@@ -0,0 +1,87 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+
+export default class DiscourseTagBanners extends Component {
+  @service store;
+  @service router;
+  @service site;
+  @tracked tag = null;
+  @tracked keepDuringLoadingRoute = false;
+  @tracked isIntersection = false;
+
+  get isVisible() {
+    const route = this.router?.currentRoute;
+
+    if (
+      route &&
```

Comment:
you seem to mostly use `this.router.currentRoute.params` so you coudl do a getter:

```javascript
get currentRouteParams() {
  return this.router?.currentRoute?.params;
}

```